pyccel/codegen/compiling/basic.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+Module handling classes for compiler information relevant to a given object
+"""
+import os
+import sys
+from filelock import FileLock
+
+class CompileObj:
+    """
+    Class containing all information necessary for compiling
+
+    Parameters
+    ----------
+    file_name     : str
+                    Name of file to be compiled
+
+    folder        : str
+                    Name of the folder where the file is found
+
+    is_module     : bool
+                    Indicates whether we are compiling a module or a program
+                    Default : True
+
+    flags         : str
+                    Any non-default flags passed to the compiler
+
+    includes      : iterable of strs
+                    include directories paths
+
+    libs          : iterable of strs
+                    required libraries
+
+    libdirs       : iterable of strs
+                    paths to directories containing the required libraries
+
+    dependencies  : iterable of CompileObjs
+                    objects which must also be compiled in order to compile this module/program
+
+    accelerators  : str
+                    Tool used to accelerate the code (e.g. openmp openacc)
+    """
+    def __init__(self,
+                 file_name,
+                 folder,
+                 is_module    = True,
+                 flags        = (),
+                 includes     = (),
+                 libs         = (),
+                 libdirs      = (),
+                 dependencies = (),
+                 accelerators = ()):
+        if not all(isinstance(d, CompileObj) for d in dependencies):
+            raise TypeError("Dependencies require necessary compile information")
```

Comment:
I think we don't need this check, we are doing it in the `add_dependencies` function.

Reply:
I agree. Fixed in a114d1cc